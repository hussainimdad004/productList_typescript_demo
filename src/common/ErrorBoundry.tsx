import * as React from 'react';
import {View, Text} from 'react-native'

interface State {
 error?: any,
 errorInfo: any
}
interface Props {

}
export default class ErrorBoundary extends React.Component<Props, State> {
    constructor(props:Props){
        super(props)
        this.state={
            error: null, errorInfo: null 
        }
    }

    componentDidCatch(error: any, errorInfo: any) {
      this.setState({
        error,
        errorInfo
      });
    }

    render() {
      if (this.state.errorInfo) {
        return (
          <View>
            <Text>Something went wrong.</Text>
            <View style={{ flex:1 }}>
              {this.state.error && this.state.error.toString()}
              <br />
              {this.state.errorInfo.componentStack}
            </View>
          </View>
        );
      }

      return this.props.children;
    }
  }