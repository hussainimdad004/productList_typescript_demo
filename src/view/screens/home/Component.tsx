import * as React from 'react';
import { View, FlatList, Image, Text, TouchableOpacity } from 'react-native';

import styles from './styles';
import { SECONDARY_DISABLE_GRAY, MAIN_RED, MAIN_WHITE, GRAY_OTHER, SECONDARY_BLACK, SECONDARY_COOL_GRAY_2, SECONDARY_COOL_GRAY, MAIN_GRAY } from '../../../constants/colors';
import WebApi from '../../../config/index'
import { scale, verticalScale } from '../../../common/scalling';
import { formatCurrency } from '../../../common/formatNumber';
import AutoHeightImage from 'react-native-auto-height-image';
import Modal from 'react-native-modal';
import { Icon } from 'react-native-elements';


export interface Product {
  name: string;
  colour: string,
  price: number,
  img: string,
  qty: number,
  id: number,
  index: number
}

interface State {
  name: string;
  _data: Array<Product>;
  local_data: Array<Product>;
  quantity: number;
  total_cart_amount: number;
  filterModal: boolean;
  filtersArray: Array<string>;
  filter_value: string;
}

class Home extends React.PureComponent<Product, State> {
  constructor(props: Product) {
    super(props);
    this.state = {
      name: props.name || 'RN + TS + RNN2',
      _data: [],
      local_data: [],
      quantity: 0,
      total_cart_amount: 0,
      filterModal: false,
      filtersArray: ['BLACK', 'RED'],
      filter_value: 'ALL'
    };
  }

  componentDidMount() {
    this.getProductList()
  }
  getProductList = () => {
    new WebApi().getProductList().then(data => {
      // console.log('dataaaaaaAAAA', data)
      data.data.forEach((item: Product) => item.qty = 0)
      this.setState({
        _data: data.data,
        local_data: data.data
      })
    })
      .catch((error) => {
        // console.log("catch error", error)
      })

  }
  removeItem = (_item: Product) => {
    this.setState({
      local_data: [...this.state.local_data],
      total_cart_amount: this.state.total_cart_amount - (_item.qty * _item.price)
    })
    _item.qty = 0
  }
  incrementQTY = (_item: Product) => {
    _item.qty += 1
    this.setState({
      local_data: [...this.state.local_data],
      total_cart_amount: this.state.total_cart_amount + _item.price
    })
  }
  decrementQTY = (_item: Product) => {
    if (_item.qty > 0) {
      _item.qty -= 1
      this.setState({
        local_data: [...this.state.local_data],
        total_cart_amount: this.state.total_cart_amount - _item.price
      })
    }
  }
  _renderRowItem = ({ item, index }: { item: Product; index: number; }) => {
    let img_url = undefined
    if (item.img) {
      img_url = item.img.replace(/^http:\/\//i, "https://")
    }
    return <TouchableOpacity
      onPress={() => {

      }}
      key={index}
      style={{
        flexDirection: "row",
        backgroundColor: MAIN_WHITE,
        marginHorizontal: scale(12),
        marginBottom: verticalScale(4),
        paddingTop: verticalScale(8),
        paddingBottom: verticalScale(6)
      }} >

      <View
        style={{
          flex: 0.17,
          marginLeft: scale(8)
        }}>
        {
          <AutoHeightImage
            width={scale(50)}
            source={{ uri: item.img }}
            style={{
              borderWidth: 1,
              borderColor: "#CCCCCC",
            }}
            resizeMode={"contain"}
          />
        }

      </View>
      <View
        style={{
          flex: 0.50,
          marginLeft: scale(34)
        }}
      >
        <Text
          style={{
            color: SECONDARY_BLACK,
            fontSize: scale(11),
            fontFamily: "BurlingamePro-CondBold",
          }}
        >
          {item.name}
        </Text>
        <Text
          style={{
            color: MAIN_RED,
            fontSize: scale(11),
            fontFamily: "BurlingamePro-CondSemiBold",
          }}
        >
          {`${formatCurrency(item.price)}`}
        </Text>
      </View>

      <View
        style={{
          flexDirection: 'column',
          alignItems: 'flex-start',
          justifyContent: 'flex-start',
          flex: 0.30,
          paddingHorizontal: scale(4),
        }}
      >
        <View
          style={{
            flexDirection: 'row',
            alignItems: 'flex-start',
            justifyContent: 'flex-start',
          }}
        >
          <TouchableOpacity
            onPress={() => {
              this.decrementQTY(item)
            }}
            style={{
              backgroundColor: MAIN_WHITE,
              borderColor: SECONDARY_COOL_GRAY_2,
              borderWidth: 1,
              borderRadius: 2,
              width: scale(25),
              alignItems: 'center',
              height: verticalScale(23)
            }}
          >
            <Text
              style={{
                fontSize: scale(18),
                fontFamily: "BurlingamePro-CondBold",
                color: SECONDARY_BLACK
              }}>
              -
            </Text>
          </TouchableOpacity>
          <Text
            style={{
              fontSize: scale(12),
              fontFamily: "BurlingamePro-CondBold",
              marginHorizontal: scale(8),
              marginTop: verticalScale(4),
              color: "#858585",
              // color: SECONDARY_COOL_GRAY_2
            }}
          >
            {item.qty}
          </Text>
          <TouchableOpacity
            onPress={() => {
              this.incrementQTY(item)
            }}
            style={{
              backgroundColor: MAIN_WHITE,
              borderColor: SECONDARY_COOL_GRAY_2,
              borderWidth: 1,
              borderRadius: 2,
              width: scale(25),
              alignItems: 'center',
              height: verticalScale(23)
            }}
          >
            <Text
              style={{
                fontSize: scale(18),
                fontFamily: "BurlingamePro-CondBold",
                color: SECONDARY_BLACK
              }}>
              +
            </Text>
          </TouchableOpacity>
        </View>

        <TouchableOpacity
          onPress={() => {
            this.removeItem(item)
          }}
          style={{
            flex: 1,
            alignSelf: "center",
            justifyContent: 'flex-end',
          }}
        >
          <Text style={{
            color: MAIN_RED,
            fontSize: scale(11),
            fontFamily: "BurlingamePro-CondSemiBold",
          }}>REMOVE</Text>
        </TouchableOpacity>
      </View>

    </TouchableOpacity>
  }

  drawLine = () => {
    return (
      <View style={{
        marginHorizontal: scale(24),
        height: 0.7,
        backgroundColor: "#aaa"
      }}>
      </View>
    );
  }
  createMainTotal = (description: string, amount: number) => {
    return (
      <View
        style={{
          marginHorizontal: scale(22),
          marginTop: verticalScale(12),
          marginBottom: verticalScale(28),
          flexDirection: "row",
          justifyContent: "space-between"
        }}
      >
        <View
          style={{
            flexDirection: 'row'
          }}
        >
          <Text
            style={{
              color: SECONDARY_BLACK,
              fontSize: scale(12),
              fontFamily: "BurlingamePro-CondBold",
            }}
          >{description}
          </Text>

        </View>
        <Text
          style={{
            color: SECONDARY_BLACK,
            fontSize: scale(12),
            fontFamily: "BurlingamePro-CondBold",
          }}
        >
          {formatCurrency(amount)}
        </Text>
      </View>
    );
  }
  filterSection = () => {
    return (
      <View style={{
        flexDirection: 'row',
        marginBottom: verticalScale(10),
        marginHorizontal: scale(16)
      }}>
        <TouchableOpacity style={{
          backgroundColor: MAIN_WHITE,
          width: '48%',
          marginRight: scale(4),
          flexDirection: "row",
          justifyContent: "space-between",
          paddingTop: verticalScale(2),
          paddingBottom: verticalScale(2),
          borderWidth: 1,
          borderColor: MAIN_RED,
          paddingHorizontal: scale(12)
        }}
          onPress={() => this.hideModal()}>
          <View style={{ flexDirection: "column", justifyContent: "space-between" }}>
            <Text style={{
              color: GRAY_OTHER,
              fontSize: scale(10),
              fontFamily: "BurlingamePro-CondSemiBold",
            }}>
              FILTERS
                </Text>

            <Text style={{
              color: MAIN_RED,
              fontSize: scale(10),
              paddingTop: verticalScale(2),
              fontFamily: "BurlingamePro-CondSemiBold"
            }}>
              {this.state.filter_value}
            </Text>

          </View>
          <Icon
            color={MAIN_RED}
            name="chevron-down"
            type="entypo"
            size={24}
            activeOpacity={0.6}
          />

        </TouchableOpacity>
      </View>

    )
  }
  _modalView = () => {
    return (
      <Modal
        style={{
          justifyContent: "flex-end",
          margin: 0,
          marginHorizontal: scale(16)
        }}
        useNativeDriver={true}
        backdropTransitionOutTiming={0}
        animationInTiming={500}
        animationOutTiming={500}
        isVisible={this.state.filterModal}
        onBackdropPress={() => {
          this.hideModal()
        }}
      >
        <View
          // maxHeight="95%"
          style={{
            justifyContent: "center",
            backgroundColor: MAIN_WHITE,
            paddingTop: verticalScale(4),
            alignItems: 'center'
          }}
        >
          <Text
            style={{
              marginTop: verticalScale(12),
              fontSize: scale(14),
              color: MAIN_RED,
              fontFamily: "BurlingamePro-CondBold",
              alignSelf: "center",
              paddingHorizontal: scale(32),
              textAlign: "center",
            }}
          >
            FILTERS
              </Text>
          <View style={{ width: '100%', }}>
            <FlatList
              style={{ marginTop: scale(8) }}
              disableVirtualization={false}
              data={this.state.filtersArray}
              renderItem={this._renderCategoryData}
              keyExtractor={item => item.toString()}
            />

          </View>
          <TouchableOpacity
            style={{
              backgroundColor: SECONDARY_COOL_GRAY,
              height: verticalScale(40),
              paddingLeft: scale(8),
              paddingRight: scale(8),
              width: "100%",
              alignItems: "center",
              flexDirection: "row",
              justifyContent: "center"
            }}
            onPress={() => {
              this.setState({
                filter_value: 'All',
                local_data: this.state._data
              })
              // this.filterList("all", this.state.sort)
              this.hideModal()
            }}
          >
            <Text
              style={{
                color: MAIN_RED,
                fontSize: scale(11),
                fontFamily: "BurlingamePro-CondBold",
                elevation: 8,
              }}
            >
              DISPLAY ALL LISTING
                      </Text>

          </TouchableOpacity>
        </View>
      </Modal>
    )
  }
  filterList = (name: string) => {
    let _name = name.toLowerCase()

    this.setState({
      local_data: this.state._data.filter((obj) => {
        return obj.colour.toLowerCase() === _name
      })
    })
  }

  _renderCategoryData = ({ item, index }: { item: string; index: number; }) => {

    return (
      <TouchableOpacity
        key={index}
        onPress={() => {
          this.hideModal()
          this.filterList(item)
          this.setState({
            filter_value: item
          })
        }}
        style={{
          justifyContent: "center",
          height: verticalScale(32.7),
          // borderTopColor: item.index == 0 ? MAIN_WHITE : '#aaa',
          // borderTopWidth: item.index == 0 ? 0 : scale(0.8),
        }}
      >
        <Text
          style={{
            color: MAIN_GRAY,
            fontSize: scale(12),
            fontFamily: "BurlingamePro-CondBold",
            textAlign: 'center'
          }}
        >
          {item}
        </Text>
      </TouchableOpacity>
    )

  }
  hideModal = () => {
    this.setState({
      filterModal: !this.state.filterModal,
    })
  }


  render() {
    return (
        <View style={styles.container}>
          <View style={{ flex: 1, marginHorizontal: scale(10), marginTop: verticalScale(60) }}>
            {this.filterSection()}
            <FlatList
              data={this.state.local_data}
              style={{ paddingVertical: 5 }}
              extraData={this.state}
              renderItem={this._renderRowItem}
              keyExtractor={item => item.id.toString()}
            />
          </View>
          {this.drawLine()}

          {this.createMainTotal("TOTAL", this.state.total_cart_amount)}
          {this._modalView()}
        </View>
    );
  }
}

export default Home;
