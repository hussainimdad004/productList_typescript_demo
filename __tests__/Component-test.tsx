import React from "react";
import Component from "../src/view/screens/home/Component";
import renderer from "react-test-renderer";
import ErrorBoundry from "../src/common/ErrorBoundry";
beforeEach(()=>{
    jest.useFakeTimers()
})
it("testing component", async () => {
    const { toJSON } = renderer.create(<ErrorBoundry><Component name={"Imdad"} colour={"#fff"} price={1} img="" qty={1} id={1} index={1} /></ErrorBoundry>)
    expect(toJSON()).toMatchSnapshot()
})