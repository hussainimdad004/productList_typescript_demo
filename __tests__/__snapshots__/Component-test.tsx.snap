// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeeee",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginHorizontal": 23.4375,
        "marginTop": 140.91549295774647,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 23.485915492957744,
          "marginHorizontal": 37.5,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderColor": "#E51C24",
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginRight": 9.375,
            "opacity": 1,
            "paddingBottom": 4.697183098591549,
            "paddingHorizontal": 28.125,
            "paddingTop": 4.697183098591549,
            "width": "48%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#666666",
                "fontFamily": "BurlingamePro-CondSemiBold",
                "fontSize": 23.4375,
              }
            }
          >
            FILTERS
          </Text>
          <Text
            style={
              Object {
                "color": "#E51C24",
                "fontFamily": "BurlingamePro-CondSemiBold",
                "fontSize": 23.4375,
                "paddingTop": 4.697183098591549,
              }
            }
          >
            ALL
          </Text>
        </View>
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            activeOpacity={0.6}
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#E51C24",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Entypo",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                ÔÖÅ
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      extraData={
        Object {
          "_data": Array [],
          "filterModal": false,
          "filter_value": "ALL",
          "filtersArray": Array [
            "BLACK",
            "RED",
          ],
          "local_data": Array [],
          "name": "Imdad",
          "quantity": 0,
          "total_cart_amount": 0,
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "paddingVertical": 5,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#aaa",
        "height": 0.7,
        "marginHorizontal": 56.25,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 65.76056338028168,
        "marginHorizontal": 51.5625,
        "marginTop": 28.183098591549296,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "fontFamily": "BurlingamePro-CondBold",
            "fontSize": 28.125,
          }
        }
      >
        TOTAL
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#333333",
          "fontFamily": "BurlingamePro-CondBold",
          "fontSize": 28.125,
        }
      }
    >
      $0.00
    </Text>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[Function]}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      pointerEvents="box-none"
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "margin": 0,
          "marginHorizontal": 37.5,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "justifyContent": "center",
            "paddingTop": 9.394366197183098,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#E51C24",
              "fontFamily": "BurlingamePro-CondBold",
              "fontSize": 32.8125,
              "marginTop": 28.183098591549296,
              "paddingHorizontal": 75,
              "textAlign": "center",
            }
          }
        >
          FILTERS
        </Text>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <RCTScrollView
            data={
              Array [
                "BLACK",
                "RED",
              ]
            }
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            style={
              Object {
                "marginTop": 18.75,
              }
            }
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "height": 76.79894366197183,
                      "justifyContent": "center",
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#817979",
                        "fontFamily": "BurlingamePro-CondBold",
                        "fontSize": 28.125,
                        "textAlign": "center",
                      }
                    }
                  >
                    BLACK
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
                style={null}
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "height": 76.79894366197183,
                      "justifyContent": "center",
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#817979",
                        "fontFamily": "BurlingamePro-CondBold",
                        "fontSize": 28.125,
                        "textAlign": "center",
                      }
                    }
                  >
                    RED
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#e3e3e3",
              "flexDirection": "row",
              "height": 93.94366197183098,
              "justifyContent": "center",
              "opacity": 1,
              "paddingLeft": 18.75,
              "paddingRight": 18.75,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#E51C24",
                "elevation": 8,
                "fontFamily": "BurlingamePro-CondBold",
                "fontSize": 25.78125,
              }
            }
          >
            DISPLAY ALL LISTING
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
